miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -342,6 +346,10 @@ end
 #!   - If ASSET is a fungible asset, then ASSET' is the total fungible asset in the account vault
 #!     after ASSET was added to it.
 export.account_vault_add_asset
+    # TODO: we execute `push.1 drop` before `emit` as decorators are not supported without other
+    # instructions - see: https://github.com/0xPolygonMiden/miden-vm/issues/1122
+    push.1 drop emit.ACCOUNT_VAULT_BEFORE_ADD_ASSET_EVENT 
```

Comment:
What about just moving this after the `exec.authenticate_account_origin` instead?

Reply:
For now it's just a placeholder, so I think that whether this call needs to be moved will depend on how it will be used (that's why for now I placed it at the beginning). But I don't have strong preference, so I can move it, no problem.